setup.py

```diff
@@ -1,25 +1,96 @@
-    'built ALE using CMake.'%(ale_c_lib))
-  sys.exit()
-
-module1 = Extension('ale_python_interface.ale_c_wrapper',
-                    libraries = ['ale_c'],
-                    include_dirs = ['src'],
-                    library_dirs = ['ale_python_interface'],
-                    extra_compile_args=['-D__STDC_CONSTANT_MACROS', '-std=c++11'],
-                    sources=['ale_python_interface/ale_c_wrapper.cpp'])
-setup(name = 'ale_python_interface',
-      version='0.6',
-      description = 'Arcade Learning Environment Python Interface',
-      url='https://github.com/bbitmaster/ale_python_interface',
-      author='Ben Goodrich',
-      license='GPL',
-      ext_modules = [module1],
-      packages=['ale_python_interface'],
-      package_dir={'ale_python_interface': 'ale_python_interface'},
-      package_data={'ale_python_interface': ['libale_c.so']})
+from setuptools import setup, Extension
+from setuptools.command.build_ext import build_ext
+import subprocess
+import platform
+import os
+
+
+class CMakeExtension(Extension):
+    def __init__(self, name, sourcedir="", config=[]):
+        Extension.__init__(self, name, sources=[])
+        self.sourcedir = os.path.abspath(sourcedir)
+        self.config = config
+
+
+class CMakeBuild(build_ext):
+    def build_extensions(self):
+        try:
+            subprocess.check_output(["cmake", "--version"])
+        except OSError:
+            raise RuntimeError(
+                "CMake must be installed to build the extensions: %s"
+                % ", ".join(ext.name for ext in self.extensions)
+            )
+
+        for ext in self.extensions:
+            extdir = os.path.abspath(os.path.dirname(self.get_ext_fullpath(ext.name)))
+            cfg = "Debug" if self.debug else "Release"
+
+            cmake_config_args = [
+                "-DOUTPUT_NAME={}".format(ext.name),
+                "-DCMAKE_BUILD_TYPE={}".format(cfg),
+                "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}".format(cfg.upper(), extdir),
+                "-DCMAKE_RUNTIME_OUTPUT_DIRECTORY_{}={}".format(cfg.upper(), extdir),
+                "-DCMAKE_ARCHIVE_OUTPUT_DIRECTORY_{}={}".format(
+                    cfg.upper(), self.build_temp
+                ),
+            ] + ext.config
+
+            cmake_build_args = []
+
+            # -DCMAKE_BUILD_TYPE doesn't work on Windows
+            # we need to specify --config Release at build type
+            if platform.system() == "Windows":
+                cmake_build_args += ["--config", cfg]
+
+            if not os.path.exists(self.build_temp):
+                os.makedirs(self.build_temp)
+
+            subprocess.check_call(
+                ["cmake", ext.sourcedir] + cmake_config_args, cwd=self.build_temp
+            )
+            subprocess.check_call(
+                ["cmake", "--build", "."] + cmake_build_args, cwd=self.build_temp
+            )
+
+
+with open("README.md", "r") as f:
```

Comment:
Does this work if you do `pip install -e Arcade-Learning-Environment`? What about `python Arcade-Learning-Environment/setup.py develop`?  I noticed we do relative paths here: https://github.com/openai/atari-py/blob/master/setup.py#L7

Reply:
Ahh yes, this will be a problem for reading the README.